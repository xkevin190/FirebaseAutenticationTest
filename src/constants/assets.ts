export const FONTS = {
  LATO: 'Lato',
};

export const ICONS = {
  FA5: {
    LOCATION_PIN: 'street-view',
    SQUARE_MESSAGE: 'comment-alt',
    CIRCLE: 'circle',
    ALERT: 'exclamation-triangle',
    EYE: 'eye',
    EYE_SLASH: 'eye-slash',
  },
  FONTISTO: {
    CLOSE: 'close-a',
  },
};
