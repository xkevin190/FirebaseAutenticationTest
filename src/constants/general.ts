export const SCREEN_ANIMATION = 'slide_from_right';
