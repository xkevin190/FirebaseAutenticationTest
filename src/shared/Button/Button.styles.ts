import {StyleSheet, ViewStyle} from 'react-native';

const basicContainer: ViewStyle = {
  justifyContent: 'center',
  alignItems: 'center',
  height: 50,
  borderRadius: 25,
};

const styles = StyleSheet.create({
  md: {
    ...basicContainer,
    width: '45%',
  } as ViewStyle,
  xxl: {
    ...basicContainer,
    width: '100%',
  } as ViewStyle,
});

export default styles;
