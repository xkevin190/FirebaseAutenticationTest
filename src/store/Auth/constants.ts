export type IAccountActionTypes = {
  CREATE_ACCOUNT: string;
  LOGIN: string;
  ClOSE_SESSION: string;
};

export const AccounthActionsTypes: IAccountActionTypes = {
  CREATE_ACCOUNT: 'CREATE_ACCOUNT',
  LOGIN: ' LOGIN',
  ClOSE_SESSION: 'ClOSE_SESSION',
};
