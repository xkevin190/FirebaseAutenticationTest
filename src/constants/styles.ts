import { FONTS } from "./assets";

export const COLORS = {
    WHITE: '#FFFFFF',
    BLACK: '#000000',
    BLACK_025: '#00000040',
    GRAY_DEFAULT: '#46484B',
    GRAY_DEFAULT_DARKER: '#646565',
    GRAY_MEDIUM: '#9DA5B9',
    GRAY_LIGHT: '#F6F8FF',
    GRAY_LIGHT_DARKER: '#DDDDDD',
    DISABLED_DEFAULT: '#DFE6ED',
    OPYA_BLUE: '#2162E3',
    OPYA_VIOLET: '#1745D5',
    MAIN_DEFAULT: '#313342',
    SECONDARY_LIGHT: '#62698B',
    ALERT_DEFAULT: '#D15686',
    ALERT_LIGHT: '#FFD0E2',
  };

  const FONTS_TYPES = {
    BLACK: 'Black',
    BLACK_ITALIC: 'BlackItalic',
    BOLD: 'Bold',
    BOLD_ITALIC: 'BoldItalic',
    HAIR_LINE: 'HairLine',
    HAIR_LINE_ITALIC: 'HairLineItalic',
    HEAVY: 'Heavy',
    HEAVY_ITALIC: 'HeavyItalic',
    ITALIC: 'Italic',
    LIGHT: 'Light',
    LIGHT_ITALIC: 'LightItalic',
    MEDIUM: 'Medium',
    MEDIUM_ITALIC: 'MediumItalic',
    REGULAR: 'Regular',
    SEMI_BOLD: 'SemiBold',
    SEMI_BOLD_ITALIC: 'SemiBoldItalic',
    THIN: 'Thin',
    THIN_ITALIC: 'ThinItalic',
  };


  export const STYLES = {
    FONT_FAMILY: {
      LATO_REGULAR: `${FONTS.LATO}-${FONTS_TYPES.REGULAR}`,
      LATO_BOLD: `${FONTS.LATO}-${FONTS_TYPES.BOLD}`,
    },
    FONT_SIZE: {
      PARAGRAPH: 14,
      SMALL_PARAGRAPH: 12,
      MEDIUM_PARAGRAPH: 16,
      INTERMEDIUM_PARAGRAPH: 18,
      LARGE_PARAGRAPH: 20,
    },
  };
  

