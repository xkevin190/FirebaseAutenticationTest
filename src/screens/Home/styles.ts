import {StyleSheet, ViewStyle} from 'react-native';

import {COLORS} from '../../constants/styles';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'column',
    backgroundColor: COLORS.GRAY_LIGHT,
  } as ViewStyle,
});

export default styles;
