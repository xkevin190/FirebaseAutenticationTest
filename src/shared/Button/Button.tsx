import React from 'react';
import {TouchableOpacity} from 'react-native';

import CustomText from '../CustomText';
import styles from './Button.styles';
import {COLORS} from '../../constants/styles';
export type ButtonSize = 'md' | 'xxl';

export interface IButtonProps {
  buttonText: string;
  testID?: string;
  size?: ButtonSize;
  rounded?: boolean;
  color?: string;
  textColor?: string;
  disabled?: boolean;
  onPress: () => void;
}

const Button: React.FC<IButtonProps> = ({
  buttonText,
  testID,
  size = 'md',
  onPress,
  color = COLORS.OPYA_BLUE,
  rounded = false,
  textColor = undefined,
  disabled = false,
}) => (
  <TouchableOpacity
    style={{
      ...styles[size],
      backgroundColor: disabled ? COLORS.DISABLED_DEFAULT : color,
      ...(!rounded && {borderRadius: 0}),
    }}
    onPress={onPress}
    testID={testID}
    disabled={disabled}>
    <CustomText text={buttonText} color={disabled ? COLORS.BLACK : textColor} />
  </TouchableOpacity>
);

export default Button;
