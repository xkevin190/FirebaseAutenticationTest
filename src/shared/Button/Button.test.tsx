// Button.test.tsx
import React from 'react';
import {render, fireEvent, act, waitFor} from '@testing-library/react-native';
import Button from './Button';
import {COLORS} from '../../constants/styles';

describe('Button Component', () => {
  const onPressMock = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with default props', () => {
    const {toJSON} = render(
      <Button buttonText="Default Button" onPress={() => {}} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('does not call onPress when button is disabled', () => {
    const onPressMock = jest.fn();
    const {getByText} = render(
      <Button buttonText="Disabled" onPress={onPressMock} disabled={true} />,
    );

    waitFor(() => {
      fireEvent.press(getByText('Disabled'));
    });
    expect(onPressMock).not.toHaveBeenCalled();
  });

  it('renders with the correct text color when enabled and disabled', () => {
    const {getByText, rerender} = render(
      <Button
        buttonText="Test Text Color"
        onPress={() => {}}
        textColor="green"
      />,
    );
    expect(getByText('Test Text Color').props.style).toMatchObject({
      color: 'green',
    });

    rerender(
      <Button
        buttonText="Test Text Color"
        onPress={() => {}}
        textColor="green"
        disabled={true}
      />,
    );
    expect(getByText('Test Text Color').props.style).toMatchObject({
      color: COLORS.BLACK,
    });
  });

  it('calls onPress when button is enabled', async () => {
    const {getByText} = render(
      <Button buttonText="Press Me" onPress={onPressMock} />,
    );

    await waitFor(() => {
      fireEvent.press(getByText('Press Me'));
    });
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });

  it('does not call onPress when button is disabled', async () => {
    const {getByText} = render(
      <Button buttonText="Press Me" onPress={onPressMock} disabled={true} />,
    );
    await waitFor(() => {
      fireEvent.press(getByText('Press Me'));
    });
    expect(onPressMock).not.toHaveBeenCalled();
  });
});
