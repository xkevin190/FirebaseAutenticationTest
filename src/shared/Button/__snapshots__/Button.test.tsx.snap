// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly with default props 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2162E3",
      "borderRadius": 0,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "45%",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontFamily": "Lato-Regular",
        "fontSize": 14,
      }
    }
    testID=""
  >
    Default Button
  </Text>
</View>
`;
